x/contracts/wasmlanche/src/memory/allocations.rs

```diff
@@ -175,20 +175,23 @@ cfg_if! {
             #[test]
             #[should_panic(expected = "already accessed")]
             fn panic_on_concurrent_access() {
-                static SINGLETON: SingletonLenMap = SingletonLenMap::new();
-
-                let test = || {
-                    // run enough times to make sure there's a collision
-                    for _ in 0..10000 {
-                        SINGLETON.with_borrow(|map| {
-                            assert!(map.is_empty());
-                        });
+                fn test_with_delay(duration: Duration) -> impl FnOnce() {
+                    static SINGLETON: SingletonLenMap = SingletonLenMap::new();
+
+                    move || {
+                        std::thread::sleep(duration);
```

Comment:
i guess this allows us to test better in multithreaded environment, but wouldn't be relevant when running in our contracts. Why add then?